### ISSUE TITLE
App crashes on startup after logging in

### ISSUE THREAD

**Issue by devlocalhost** [galaxygoldfish/pineapple#4]
Steps to reproduce:
1. Login with the app (if you can, which is another issue, but can be "fixed" by logging from the "accounts" tab)
2. Close the app
3. Open app
4. App crashes

Logs:

```
Process: com.pineapple.app
PID: 31865
UID: 10265
Frozen: false
Flags: 0x20c8be44
Package: com.pineapple.app v1 (v1.0-beta)
Foreground: Yes
Process-Runtime: 78647891
Build: Xiaomi/laurel_sprout/laurel_sprout:11/RKQ1.200903.002/V12.0.26.0.RFQMIXM:user/release-keys
Loading-Progress: 1.0
Dropped-Count: 0

retrofit2.HttpException: HTTP 404 
	at retrofit2.KotlinExtensions$await$2$2.onResponse(KotlinExtensions.kt:53)
	at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:129)
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [androidx.compose.ui.platform.MotionDurationScaleImpl@8b3d23f, androidx.compose.runtime.BroadcastFrameClock@a59e70c, StandaloneCoroutine{Cancelling}@bf53855, AndroidUiDispatcher@551126a]
```

**Comment by galaxygoldfish** [comment]
I tried to replicate this issue with a clean install of the app on a Google Pixel 6a and it doesn't seem to affect my device. It could be because of dependency updates merged with the latest pull request, but if installing the app with the latest version of the code doesn't work could you please include a longer snippet of the log so we can see what line in the code caused this? Thanks!

**Comment by devlocalhost** [comment]
> I tried to replicate this issue with a clean install of the app on a Google Pixel 6a and it doesn't seem to affect my device. It could be because of dependency updates merged with the latest pull request, but if installing the app with the latest version of the code doesn't work could you please include a longer snippet of the log so we can see what line in the code caused this? Thanks!

The issue is still there for me, even after updating. This happens only once, and only after logging in. Logs:
```
FATAL EXCEPTION: main
Process: com.pineapple.app, PID: 21937
retrofit2.HttpException: HTTP 404
	at retrofit2.KotlinExtensions$await$2$2.onResponse(KotlinExtensions.kt:53)
	at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:129)
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [androidx.compose.ui.platform.MotionDurationScaleImpl@bc3403f, androidx.compose.runtime.BroadcastFrameClock@f6fbd0c, StandaloneCoroutine{Cancelling}@1755655, AndroidUiDispatcher@a01d86a]
```

**Comment by devlocalhost** [comment]
Logs were taken with the app "Logfox". That's all the app gives me. If there's a way to get more logs, as you said, let me know

**Comment by devlocalhost** [comment]
Will a debug build give more logs? Because I'm using a release build

Tried with debug build, it does not give more output

**Comment by galaxygoldfish** [comment]
Here are two ways to get full device logs:
- Use the Logcat command-line tool (https://developer.android.com/tools/logcat#:~:text=Command-line%20syntax%20To%20run%20logcat%20through%20the%20adb,but%20that%20just%20expands%20to%20adb%20shell%20logcat)
- Open Android Studio, connect to your device with the app open, and click on the Logcat tab

**Comment by devlocalhost** [comment]
https://katb.in/adopibocozu (once again, I do not use android studio.)

```
12-04 20:06:15.501   935  3542 I ActivityTaskManager: START u0 {act=android.intent.action.VIEW cat=[android.intent.category.BROWSABLE] dat=pineapple://login/... flg=0x14000000 cmp=com.pineapple.app/.MainActivity (has extras)} from uid 10162
12-04 20:06:15.502   935  3542 D CompatibilityChangeReporter: Compat change id reported: 265464455; UID 10238; state: DISABLED
12-04 20:06:15.502   935  3542 D CompatibilityChangeReporter: Compat change id reported: 265452344; UID 10238; state: DISABLED
12-04 20:06:15.502   935  3542 D CompatibilityChangeReporter: Compat change id reported: 266124927; UID 10238; state: DISABLED
12-04 20:06:15.503   935  3542 D CompatibilityChangeReporter: Compat change id reported: 265451093; UID 10238; state: DISABLED
12-04 20:06:15.503   935  3542 D CompatibilityChangeReporter: Compat change id reported: 265456536; UID 10238; state: DISABLED
12-04 20:06:15.503   935  3542 D CompatibilityChangeReporter: Compat change id reported: 255940284; UID 10238; state: DISABLED
12-04 20:06:15.503   935  3542 D CompatibilityChangeReporter: Compat change id reported: 236283604; UID 10238; state: DISABLED
12-04 20:06:15.503   935  3542 D CompatibilityChangeReporter: Compat change id reported: 263959004; UID 10238; state: DISABLED
12-04 20:06:15.503   935  3542 D CompatibilityChangeReporter: Compat change id reported: 264304459; UID 10238; state: DISABLED
12-04 20:06:15.503   935  3542 D CompatibilityChangeReporter: Compat change id reported: 264301586; UID 10238; state: DISABLED
12-04 20:06:15.503   935  3542 D CompatibilityChangeReporter: Compat change id reported: 254631730; UID 10238; state: DISABLED
12-04 20:06:15.503   935  3542 D CompatibilityChangeReporter: Compat change id reported: 273509367; UID 10238; state: DISABLED
12-04 20:06:15.503   935  3542 D CompatibilityChangeReporter: Compat change id reported: 263259275; UID 10238; state: DISABLED
12-04 20:06:15.504   935  3542 D CompatibilityChangeReporter: Compat change id reported: 194480991; UID 10238; state: ENABLED
12-04 20:06:15.505   935  3542 I ActivityTaskManager: Launching r: ActivityRecord{32c9e75 u0 com.pineapple.app/.MainActivity} from background: ActivityRecord{455b277 u0 com.android.chrome/com.google.android.apps.chrome.Main} t2023}. New task: false
12-04 20:06:15.513   935  3542 D CompatibilityChangeReporter: Compat change id reported: 184838306; UID 10238; state: DISABLED
12-04 20:06:15.513   935  3542 D CompatibilityChangeReporter: Compat change id reported: 185004937; UID 10238; state: DISABLED
12-04 20:06:15.517   935  3542 V SplashScreenExceptionList: SplashScreen checking exception for package com.pineapple.app (target sdk:33) -> false
12-04 20:06:15.518   935  3542 D CompatibilityChangeReporter: Compat change id reported: 205907456; UID 10238; state: ENABLED
12-04 20:06:15.548   935  5769 D CoreBackPreview: Window{90f1c98 u0 Splash Screen com.pineapple.app}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@a7240d6, mPriority=0}
12-04 20:06:15.564   935  3542 W ActivityTaskManager: Tried to set launchTime (0) < mLastActivityLaunchTime (2245402)
12-04 20:06:15.617   935  5769 D CoreBackPreview: Window{6c5dc48 u0 com.pineapple.app/com.pineapple.app.MainActivity}: Setting back callback null
12-04 20:06:15.626   935  1784 W UsageStatsService: Unexpected activity event reported! (com.pineapple.app/com.pineapple.app.MainActivity event : 23 instanceId : 236564969)
12-04 20:06:15.663   935  3542 D CoreBackPreview: Window{3f0824f u0 com.pineapple.app/com.pineapple.app.MainActivity}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@3e30ee5, mPriority=0}
12-04 20:06:15.709   935  3542 I ActivityTaskManager: START u0 {act=android.intent.action.VIEW cat=[android.intent.category.BROWSABLE] dat=pineapple://login/... flg=0x1440c000 cmp=com.pineapple.app/.MainActivity (has extras)} from uid 10238
12-04 20:06:15.719   935  3542 V SplashScreenExceptionList: SplashScreen checking exception for package com.pineapple.app (target sdk:33) -> false
12-04 20:06:15.724   935  3542 W ActivityTaskManager: Duplicate finish request for r=ActivityRecord{32c9e75 u0 com.pineapple.app/.MainActivity} t2024 f}}
12-04 20:06:15.761   935  1051 W ActivityTaskManager: Tried to set launchTime (0) < mLastActivityLaunchTime (2275632)
12-04 20:06:15.790   935  3542 D CoreBackPreview: Window{663da12 u0 com.pineapple.app/com.pineapple.app.MainActivity}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@7635be0, mPriority=0}
12-04 20:06:16.165   935  1051 D CoreBackPreview: Window{90f1c98 u0 Splash Screen com.pineapple.app EXITING}: Setting back callback null
12-04 20:06:16.176   935  1051 D CoreBackPreview: Window{1183dd6 u0 com.android.chrome/com.google.android.apps.chrome.Main}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@ea6311a, mPriority=-1}
12-04 20:06:16.725   935  1051 D CoreBackPreview: Window{3f0824f u0 com.pineapple.app/com.pineapple.app.MainActivity}: Setting back callback null
12-04 20:06:17.027 14976 14976 E AndroidRuntime: FATAL EXCEPTION: main
12-04 20:06:17.027 14976 14976 E AndroidRuntime: Process: com.pineapple.app, PID: 14976
12-04 20:06:17.027 14976 14976 E AndroidRuntime: retrofit2.HttpException: HTTP 404 
12-04 20:06:17.027 14976 14976 E AndroidRuntime: 	at retrofit2.KotlinExtensions$await$2$2.onResponse(KotlinExtensions.kt:53)
12-04 20:06:17.027 14976 14976 E AndroidRuntime: 	at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:129)
12-04 20:06:17.027 14976 14976 E AndroidRuntime: 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
12-04 20:06:17.027 14976 14976 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
12-04 20:06:17.027 14976 14976 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
12-04 20:06:17.027 14976 14976 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:1012)
12-04 20:06:17.027 14976 14976 E AndroidRuntime: 	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [androidx.compose.ui.platform.MotionDurationScaleImpl@9bb44b3, androidx.compose.runtime.BroadcastFrameClock@91cfa70, StandaloneCoroutine{Cancelling}@844f1e9, AndroidUiDispatcher@a462e6e]
12-04 20:06:17.032   935  3542 W ActivityTaskManager:   Force finishing activity com.pineapple.app/.MainActivity
12-04 20:06:17.032   935 22728 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
12-04 20:06:17.046   935  1787 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
12-04 20:06:17.046   935  1787 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
12-04 20:06:17.081   935  5333 I WindowManager: WIN DEATH: Window{663da12 u0 com.pineapple.app/com.pineapple.app.MainActivity}
12-04 20:06:17.082   935  3542 I ActivityManager: Process com.pineapple.app (pid 14976) has died: fg  TOP 
12-04 20:06:17.121   935  1734 W WindowManager: Failed to deliver inset state change to w=Window{663da12 u0 com.pineapple.app/com.pineapple.app.MainActivity EXITING}
12-04 20:06:17.121   935  1734 W WindowManager: android.os.DeadObjectException
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.os.BinderProxy.transactNative(Native Method)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.os.BinderProxy.transact(BinderProxy.java:584)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.view.IWindow$Stub$Proxy.insetsControlChanged(IWindow.java:473)
12-04 20:06:17.121   935  1734 W WindowManager: 	at com.android.server.wm.WindowState.notifyInsetsControlChanged(WindowState.java:4011)
12-04 20:06:17.121   935  1734 W WindowManager: 	at com.android.server.wm.InsetsStateController.lambda$notifyPendingInsetsControlChanged$4(InsetsStateController.java:352)
12-04 20:06:17.121   935  1734 W WindowManager: 	at com.android.server.wm.InsetsStateController.$r8$lambda$An2IoiA3BeA5IWc6QwBOjKArM80(Unknown Source:0)
12-04 20:06:17.121   935  1734 W WindowManager: 	at com.android.server.wm.InsetsStateController$$ExternalSyntheticLambda3.run(Unknown Source:2)
12-04 20:06:17.121   935  1734 W WindowManager: 	at com.android.server.wm.WindowAnimator.executeAfterPrepareSurfacesRunnables(WindowAnimator.java:345)
12-04 20:06:17.121   935  1734 W WindowManager: 	at com.android.server.wm.WindowAnimator.animate(WindowAnimator.java:226)
12-04 20:06:17.121   935  1734 W WindowManager: 	at com.android.server.wm.WindowAnimator.lambda$new$1(WindowAnimator.java:106)
12-04 20:06:17.121   935  1734 W WindowManager: 	at com.android.server.wm.WindowAnimator.$r8$lambda$fo2Nk5bb9hY3lvmHsTnTEwtZMbI(Unknown Source:0)
12-04 20:06:17.121   935  1734 W WindowManager: 	at com.android.server.wm.WindowAnimator$$ExternalSyntheticLambda1.doFrame(Unknown Source:2)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1229)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1239)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.view.Choreographer.doCallbacks(Choreographer.java:899)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.view.Choreographer.doFrame(Choreographer.java:827)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1214)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.os.Handler.handleCallback(Handler.java:942)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.os.Handler.dispatchMessage(Handler.java:99)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.os.Looper.loopOnce(Looper.java:201)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.os.Looper.loop(Looper.java:288)
12-04 20:06:17.121   935  1734 W WindowManager: 	at android.os.HandlerThread.run(HandlerThread.java:67)
12-04 20:06:17.121   935  1734 W WindowManager: 	at com.android.server.ServiceThread.run(ServiceThread.java:44)
12-04 20:06:17.134   935  1734 W WindowManager: Exception thrown during dispatchAppVisibility Window{663da12 u0 com.pineapple.app/com.pineapple.app.MainActivity EXITING}
12-04 20:06:17.134   935  1734 W WindowManager: android.os.DeadObjectException
12-04 20:06:17.134   935  1734 W WindowManager: 	at android.os.BinderProxy.transactNative(Native Method)
12-04 20:06:17.134   935  1734 W WindowManager: 	at android.os.BinderProxy.transact(BinderProxy.java:584)
12-04 20:06:17.134   935  1734 W WindowManager: 	at android.view.IWindow$Stub$Proxy.dispatchAppVisibility(IWindow.java:536)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.WindowState.sendAppVisibilityToClients(WindowState.java:3478)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.WindowContainer.sendAppVisibilityToClients(WindowContainer.java:1234)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.WindowToken.setClientVisible(WindowToken.java:392)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.ActivityRecord.setClientVisible(ActivityRecord.java:6811)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.ActivityRecord.onAnimationFinished(ActivityRecord.java:7633)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.ActivityRecord.postApplyAnimation(ActivityRecord.java:5494)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.ActivityRecord.commitVisibility(ActivityRecord.java:5454)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.ActivityRecord.commitVisibility(ActivityRecord.java:5458)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.AppTransitionController.handleClosingApps(AppTransitionController.java:1194)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.AppTransitionController.handleAppTransitionReady(AppTransitionController.java:304)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.RootWindowContainer.checkAppTransitionReady(RootWindowContainer.java:970)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.RootWindowContainer.performSurfacePlacementNoTrace(RootWindowContainer.java:834)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.RootWindowContainer.performSurfacePlacement(RootWindowContainer.java:777)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.WindowSurfacePlacer.performSurfacePlacementLoop(WindowSurfacePlacer.java:177)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.WindowSurfacePlacer.performSurfacePlacement(WindowSurfacePlacer.java:126)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.WindowSurfacePlacer.performSurfacePlacement(WindowSurfacePlacer.java:115)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.wm.WindowSurfacePlacer$Traverser.run(WindowSurfacePlacer.java:57)
12-04 20:06:17.134   935  1734 W WindowManager: 	at android.os.Handler.handleCallback(Handler.java:942)
12-04 20:06:17.134   935  1734 W WindowManager: 	at android.os.Handler.dispatchMessage(Handler.java:99)
12-04 20:06:17.134   935  1734 W WindowManager: 	at android.os.Looper.loopOnce(Looper.java:201)
12-04 20:06:17.134   935  1734 W WindowManager: 	at android.os.Looper.loop(Looper.java:288)
12-04 20:06:17.134   935  1734 W WindowManager: 	at android.os.HandlerThread.run(HandlerThread.java:67)
12-04 20:06:17.134   935  1734 W WindowManager: 	at com.android.server.ServiceThread.run(ServiceThread.java:44)
12-04 20:06:17.150   935  1731 W ActivityManager: setHasOverlayUi called on unknown pid: 14976
12-04 20:06:17.534   935  1731 W ActivityTaskManager: Activity top resumed state loss timeout for ActivityRecord{b4edbc8 u0 com.pineapple.app/.MainActivity} t-1 f}}
12-04 20:06:18.618   935  1721 D CompatibilityChangeReporter: Compat change id reported: 173031413; UID 1000; state: ENABLED
12-04 20:06:18.618   935  1721 D CompatibilityChangeReporter: Compat change id reported: 173031413; UID 1000; state: DISABLED
12-04 20:06:18.846   935  1054 D CoreBackPreview: Window{47b0fd u0 VolumeDialogImpl}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@bd5e043, mPriority=0}
12-04 20:06:20.146   935  1054 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.pineapple.app/.MainActivity bnds=[521,207][682,411]} from uid 10226
12-04 20:06:20.157   935  1054 V SplashScreenExceptionList: SplashScreen checking exception for package com.pineapple.app (target sdk:33) -> false
12-04 20:06:20.172   935  1731 D CompatibilityChangeReporter: Compat change id reported: 135634846; UID 10238; state: DISABLED
12-04 20:06:20.172   935  1731 D CompatibilityChangeReporter: Compat change id reported: 177438394; UID 10238; state: DISABLED
12-04 20:06:20.172   935  1731 D CompatibilityChangeReporter: Compat change id reported: 135772972; UID 10238; state: DISABLED
12-04 20:06:20.172   935  1731 D CompatibilityChangeReporter: Compat change id reported: 135754954; UID 10238; state: ENABLED
12-04 20:06:20.174   935  1788 D CompatibilityChangeReporter: Compat change id reported: 143937733; UID 10238; state: ENABLED
12-04 20:06:20.182   935  1788 I ActivityManager: Start proc 14980:com.pineapple.app/u0a238 for next-top-activity {com.pineapple.app/com.pineapple.app.MainActivity}
12-04 20:06:20.184   935  5570 D CoreBackPreview: Window{d9f6033 u0 Splash Screen com.pineapple.app}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@5f3969, mPriority=0}
12-04 20:06:20.259     0     0 I binder  : 14980:14980 ioctl 40046210 7fffffc1b0 returned -22
12-04 20:06:20.365   935  5342 D CoreBackPreview: Window{47b0fd u0 VolumeDialogImpl}: Setting back callback null
12-04 20:06:20.496   935  5342 D CoreBackPreview: Window{d80bc23 u0 com.pineapple.app/com.pineapple.app.MainActivity}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@2fc4ed9, mPriority=0}
```

**Comment by BabyOilJohnson** [comment]
**Log from Logcat**

```
---------------------------- PROCESS STARTED (11251) for package com.pineapple.app ----------------------------
2025-04-02 16:40:29.420 11251-11251 LoadedApk               com.pineapple.app                    D  LoadedApk::makeApplication() appContext=android.app.ContextImpl@c69d9c0 appContext.mOpPackageName=com.pineapple.app appContext.mBasePackageName=com.pineapple.app appContext.mPackageInfo=android.app.LoadedApk@3ea66f9
2025-04-02 16:40:29.420 11251-11251 NetworkSecurityConfig   com.pineapple.app                    D  No Network Security Config specified, using platform default
2025-04-02 16:40:29.452 11251-11251 NetworkSecurityConfig   com.pineapple.app                    D  No Network Security Config specified, using platform default
2025-04-02 16:40:29.472 11251-11251 ActivityThread          com.pineapple.app                    D  handleBindApplication() -- skipGraphicsSupport=false
2025-04-02 16:40:29.548 11251-11251 DecorView               com.pineapple.app                    I  [INFO] isPopOver=false, config=true
2025-04-02 16:40:29.548 11251-11251 DecorView               com.pineapple.app                    I  updateCaptionType >> DecorView@58531ab[], isFloating=false, isApplication=true, hasWindowDecorCaption=false, hasWindowControllerCallback=true
2025-04-02 16:40:29.548 11251-11251 DecorView               com.pineapple.app                    D  setCaptionType = 0, this = DecorView@58531ab[]
2025-04-02 16:40:29.612 11251-11251 InputTransport          com.pineapple.app                    D  Input channel constructed: 'bfea80f', fd=75
2025-04-02 16:40:29.619 11251-11251 ViewRootIm...nActivity] com.pineapple.app                    I  setView = com.android.internal.policy.DecorView@58531ab TM=true
2025-04-02 16:40:29.808 11251-11251 SurfaceControl          com.pineapple.app                    I  assignNativeObject: nativeObject = 0 Surface(name=null)/@0x61b7291 / android.view.SurfaceControl.readFromParcel:1121 android.view.IWindowSession$Stub$Proxy.relayout:1840 android.view.ViewRootImpl.relayoutWindow:9005 android.view.ViewRootImpl.performTraversals:3360 android.view.ViewRootImpl.doTraversal:2618 android.view.ViewRootImpl$TraversalRunnable.run:9971 android.view.Choreographer$CallbackRecord.run:1010 android.view.Choreographer.doCallbacks:809 android.view.Choreographer.doFrame:744 android.view.Choreographer$FrameDisplayEventReceiver.run:995 
2025-04-02 16:40:29.810 11251-11251 ViewRootIm...nActivity] com.pineapple.app                    I  Relayout returned: old=(0,0,1080,2340) new=(0,0,1080,2340) req=(1080,2340)0 dur=14 res=0x7 s={true 531958583296} ch=true fn=-1
2025-04-02 16:40:29.826 11251-11581 Gralloc4                com.pineapple.app                    I  mapper 4.x is not supported
2025-04-02 16:40:29.827 11251-11581 Gralloc3                com.pineapple.app                    W  mapper 3.x is not supported
2025-04-02 16:40:29.829 11251-11581 gralloc                 com.pineapple.app                    I  Arm Module v1.0
2025-04-02 16:40:29.893 11251-11251 D                       com.pineapple.app                    E  dbbedbddsdddssddsdssddsds
2025-04-02 16:40:29.917 10429-10450 WindowManager           system_server                        E  win=Window{e89364b u0 Splash Screen com.pineapple.app EXITING} destroySurfaces: appStopped=false win.mWindowRemovalAllowed=true win.mRemoveOnExit=true win.mViewVisibility=0 caller=com.android.server.wm.ActivityRecord.destroySurfaces:5251 com.android.server.wm.ActivityRecord.destroySurfaces:5232 com.android.server.wm.WindowState.onExitAnimationDone:5657 com.android.server.wm.WindowStateAnimator.onAnimationFinished:332 com.android.server.wm.WindowState.onAnimationFinished:6122 com.android.server.wm.-$$Lambda$dwJG8BAnLlvKNGuDY9U3-haNY4M.onAnimationFinished:2 com.android.server.wm.SurfaceAnimator.lambda$getFinishedCallback$0$SurfaceAnimator:112 
2025-04-02 16:40:30.348 11251-11251 m.pineapple.ap          com.pineapple.app                    W  Accessing hidden method Landroid/graphics/Canvas;->drawPatch(Landroid/graphics/NinePatch;Landroid/graphics/Rect;Landroid/graphics/Paint;)V (greylist-max-o, linking, denied)
2025-04-02 16:40:30.348 11251-11251 m.pineapple.ap          com.pineapple.app                    W  Accessing hidden method Landroid/graphics/Canvas;->drawPatch(Landroid/graphics/NinePatch;Landroid/graphics/RectF;Landroid/graphics/Paint;)V (greylist-max-o, linking, denied)
2025-04-02 16:40:31.059 11251-11251 Choreographer           com.pineapple.app                    I  Skipped 72 frames!  The application may be doing too much work on its main thread.
2025-04-02 16:40:31.068 11251-11581 OpenGLRenderer          com.pineapple.app                    I  Davey! duration=1402ms; Flags=0, IntendedVsync=456391412455993, Vsync=456391579122653, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=456391585321879, AnimationStart=456391585326571, PerformTraversalsStart=456391853346109, DrawStart=456392703553763, SyncQueued=456392803508839, SyncStart=456392803854647, IssueDrawCommandsStart=456392804207378, SwapBuffers=456392814168032, FrameCompleted=456392815321455, DequeueBufferDuration=136769, QueueBufferDuration=534269, GpuCompleted=0, 
2025-04-02 16:40:31.152 11251-11251 TAG                     com.pineapple.app                    E  authenticating user
2025-04-02 16:40:31.171 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  --> POST https://www.reddit.com/api/v1/access_token
2025-04-02 16:40:31.171 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  Content-Type: application/x-www-form-urlencoded
2025-04-02 16:40:31.171 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  Content-Length: 153
2025-04-02 16:40:31.172 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  Authorization: Basic "Client_Secret"
2025-04-02 16:40:31.172 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  grant_type=authorization_code&User-Agent=android%3Acom.pineapple.app%3Av1.0-beta&code=vk-pfPmvCqT9kvLW3bq8kf9_JAX4gw&redirect_uri=pineapple%3A%2F%2Flogin
2025-04-02 16:40:31.173 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  --> END POST (153-byte body)
2025-04-02 16:40:31.465 11251-11581 OpenGLRenderer          com.pineapple.app                    I  Davey! duration=1616ms; Flags=0, IntendedVsync=456391595505795, Vsync=456392795505747, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=456392807116647, AnimationStart=456392807123109, PerformTraversalsStart=456393066948416, DrawStart=456393067414993, SyncQueued=456393191867993, SyncStart=456393192730993, IssueDrawCommandsStart=456393193490724, SwapBuffers=456393209591801, FrameCompleted=456393212395185, DequeueBufferDuration=388846, QueueBufferDuration=1782616, GpuCompleted=0, 
2025-04-02 16:40:31.611 11251-11251 ViewRootIm...nActivity] com.pineapple.app                    I  MSG_WINDOW_FOCUS_CHANGED 1 1
2025-04-02 16:40:31.612 11251-11251 InputMethodManager      com.pineapple.app                    D  prepareNavigationBarInfo() DecorView@58531ab[MainActivity]
2025-04-02 16:40:31.612 11251-11251 InputMethodManager      com.pineapple.app                    D  getNavigationBarColor() 0
2025-04-02 16:40:31.613 11251-11251 InputMethodManager      com.pineapple.app                    D  prepareNavigationBarInfo() DecorView@58531ab[MainActivity]
2025-04-02 16:40:31.613 11251-11251 InputMethodManager      com.pineapple.app                    D  getNavigationBarColor() 0
2025-04-02 16:40:31.613 11251-11251 InputMethodManager      com.pineapple.app                    V  Starting input: tba=com.pineapple.app ic=null mNaviBarColor 0 mIsGetNaviBarColorSuccess true , NavVisible : true , NavTrans : false
2025-04-02 16:40:31.614 11251-11251 InputMethodManager      com.pineapple.app                    D  startInputInner - Id : 0
2025-04-02 16:40:31.614 11251-11251 InputMethodManager      com.pineapple.app                    I  startInputInner - mService.startInputOrWindowGainedFocus
2025-04-02 16:40:31.624 11251-11560 InputTransport          com.pineapple.app                    D  Input channel constructed: 'ClientS', fd=83
2025-04-02 16:40:31.687 11251-11251 TAG                     com.pineapple.app                    E  authenticating user
2025-04-02 16:40:31.694 11251-11612 okhttp.OkHttpClient     com.pineapple.app                    I  --> POST https://www.reddit.com/api/v1/access_token
2025-04-02 16:40:31.695 11251-11612 okhttp.OkHttpClient     com.pineapple.app                    I  Content-Type: application/x-www-form-urlencoded
2025-04-02 16:40:31.695 11251-11612 okhttp.OkHttpClient     com.pineapple.app                    I  Content-Length: 153
2025-04-02 16:40:31.695 11251-11612 okhttp.OkHttpClient     com.pineapple.app                    I  Authorization: Basic NkU0RFNVMFpHeTUwTTJwSXg1NjFHUTo=
2025-04-02 16:40:31.696 11251-11612 okhttp.OkHttpClient     com.pineapple.app                    I  grant_type=authorization_code&User-Agent=android%3Acom.pineapple.app%3Av1.0-beta&code=vk-pfPmvCqT9kvLW3bq8kf9_JAX4gw&redirect_uri=pineapple%3A%2F%2Flogin
2025-04-02 16:40:31.696 11251-11612 okhttp.OkHttpClient     com.pineapple.app                    I  --> END POST (153-byte body)
2025-04-02 16:40:31.789 11251-11251 InputMethodManager      com.pineapple.app                    D  prepareNavigationBarInfo() DecorView@58531ab[MainActivity]
2025-04-02 16:40:31.789 11251-11251 InputMethodManager      com.pineapple.app                    D  getNavigationBarColor() 0
2025-04-02 16:40:31.789 11251-11251 InputMethodManager      com.pineapple.app                    V  Starting input: tba=com.pineapple.app ic=null mNaviBarColor 0 mIsGetNaviBarColorSuccess true , NavVisible : true , NavTrans : false
2025-04-02 16:40:31.789 11251-11251 InputMethodManager      com.pineapple.app                    D  startInputInner - Id : 0
2025-04-02 16:40:31.793 11251-11251 SurfaceControl          com.pineapple.app                    I  nativeRelease nativeObject s[531956616128]
2025-04-02 16:40:31.793 11251-11251 SurfaceControl          com.pineapple.app                    I  nativeRelease nativeObject e[531956616128]
2025-04-02 16:40:31.793 11251-11251 SurfaceControl          com.pineapple.app                    I  nativeRelease nativeObject s[531956617184]
2025-04-02 16:40:31.793 11251-11251 SurfaceControl          com.pineapple.app                    I  nativeRelease nativeObject e[531956617184]
2025-04-02 16:40:32.689 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  <-- 404 https://www.reddit.com/api/v1/access_token (1515ms)
2025-04-02 16:40:32.689 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  expires: -1
2025-04-02 16:40:32.689 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  cache-control: private, s-maxage=0, max-age=0, must-revalidate, no-store
2025-04-02 16:40:32.689 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  content-type: application/json; charset=UTF-8
2025-04-02 16:40:32.689 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  x-ua-compatible: IE=edge
2025-04-02 16:40:32.690 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  set-cookie: loid=000000001mgy10i5xs.2.1743592244531.Z0FBQUFBQm43UnMwY1pFaWIyeWFwYkk4b3UtTk5sNWFpcEJmTXAxYVE4Mkt6T0xUY2xOUUFWRlhoeHRzRTZsUGU3bi1SdFd4QnRzMmY2T2oxU3A2cDFRdmYyWWJmZDRnejk2WllTQjROQkxSNFY2SDhId0thLU1MSDNkSDlUQk5BdTd6THlGQWtnVE8; Domain=reddit.com; Max-Age=63071999; Path=/; expires=Fri, 02-Apr-2027 11:10:44 GMT; secure
2025-04-02 16:40:32.690 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  set-cookie: session_tracker=addfamgbakhiqpdirr.0.1743592244557.Z0FBQUFBQm43UnMwRWRQX0xVZXJuSzlYdGxxa3dSTEROVmhxTV94TUJzS29WOVhKZWdyTUxUSjR5dDJSTElWTGJjdkphLWRWU3BteUtOY2VOQUZlQlRqc2VJUURUOXJpUWVUV2I3cEM2VFVVR3ltNVRHY3ZrdHg1eFFJdmRLSWMyalRMbVMya0locDI; Domain=reddit.com; Max-Age=7199; Path=/; expires=Wed, 02-Apr-2025 13:10:44 GMT; secure
2025-04-02 16:40:32.690 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  accept-ranges: bytes
2025-04-02 16:40:32.690 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  date: Wed, 02 Apr 2025 11:10:44 GMT
2025-04-02 16:40:32.690 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  via: 1.1 varnish
2025-04-02 16:40:32.690 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  vary: Accept-Encoding
2025-04-02 16:40:32.690 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  strict-transport-security: max-age=31536000; includeSubdomains
2025-04-02 16:40:32.690 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  x-content-type-options: nosniff
2025-04-02 16:40:32.690 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  x-frame-options: SAMEORIGIN
2025-04-02 16:40:32.690 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  x-xss-protection: 1; mode=block
2025-04-02 16:40:32.691 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  set-cookie: edgebucket=PHN10PBqyP76Xk3ULR; Domain=reddit.com; Max-Age=63071999; Path=/;  secure
2025-04-02 16:40:32.691 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  server: snooserv
2025-04-02 16:40:32.691 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  report-to: {"group": "w3-reporting-nel", "max_age": 14400, "include_subdomains": true,  "endpoints": [{ "url": "https://w3-reporting-nel.reddit.com/reports" }]}, {"group": "w3-reporting", "max_age": 14400, "include_subdomains": true, "endpoints": [{ "url": "https://w3-reporting.reddit.com/reports" }]}, {"group": "w3-reporting-csp", "max_age": 14400, "include_subdomains": true, "endpoints": [{ "url": "https://w3-reporting-csp.reddit.com/reports" }]}
2025-04-02 16:40:32.691 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  nel: {"report_to": "w3-reporting-nel", "max_age": 14400, "include_subdomains": false, "success_fraction": 1.0, "failure_fraction": 1.0}
2025-04-02 16:40:32.692 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  {"message": "Not Found", "error": 404}
2025-04-02 16:40:32.692 11251-11607 okhttp.OkHttpClient     com.pineapple.app                    I  <-- END HTTP (38-byte body)
2025-04-02 16:40:32.721 11251-11612 okhttp.OkHttpClient     com.pineapple.app                    I  <-- HTTP FAILED: okhttp3.internal.http2.StreamResetException: stream was reset: CANCEL
2025-04-02 16:40:32.733 11251-11251 AndroidRuntime          com.pineapple.app                    E  FATAL EXCEPTION: main
                                                                                                    Process: com.pineapple.app, PID: 11251
                                                                                                    retrofit2.HttpException: HTTP 404 
                                                                                                    	at retrofit2.KotlinExtensions$await$2$2.onResponse(KotlinExtensions.kt:53)
                                                                                                    	at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:129)
                                                                                                    	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                                                                                                    	at java.lang.Thread.run(Thread.java:923)
                                                                                                    	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [androidx.compose.ui.platform.MotionDurationScaleImpl@86e01f8, androidx.compose.runtime.BroadcastFrameClock@94096d1, StandaloneCoroutine{Cancelling}@51a1536, AndroidUiDispatcher@c55ac37]
2025-04-02 16:40:32.829 11251-11251 Process                 com.pineapple.app                    I  Sending signal. PID: 11251 SIG: 9
---------------------------- PROCESS ENDED (11251) for package com.pineapple.app ----------------------------
```

**Comment by galaxygoldfish** [comment]
The codebase has undergone some major changes since 2025 which likely have solved this issue.. please try again building the latest version and feel free to reopen the issue again if this still persists. And include device type, android version too if we need to debug further
